src/Analysis/Ast/Impl/Indexing/SymbolIndexWalker.cs

```diff
@@ -0,0 +1,324 @@
+                SymbolKind.None,
+                span,
+                children: children
+            ));
+        }
+
+
+        private void AddVarSymbol(NameExpression node) {
+            if (node == null) {
+                return;
+            }
+
+            var kind = SymbolKind.Variable;
+
+            switch (node.Name) {
+                case "*":
+                case "_":
+                    return;
+                case var s when (_stack.Parent == null || _stack.Parent == SymbolKind.Class) && ConstantLike.IsMatch(s):
+                    kind = SymbolKind.Constant;
+                    break;
+            }
+
+            var span = node.GetSpan(_ast);
+
+            _stack.AddSymbol(new HierarchicalSymbol(node.Name, kind, span));
+        }
+
+        private void AddVarSymbolRecursive(Expression node) {
+            if (node == null) {
+                return;
+            }
+
+            switch (node) {
+                case NameExpression ne:
+                    AddVarSymbol(ne);
+                    return;
+
+                case SequenceExpression se:
+                    foreach (var item in se.Items.MaybeEnumerate()) {
+                        AddVarSymbolRecursive(item);
+                    }
+                    return;
+            }
+        }
+
+        private (SymbolKind kind, string functionKind)? DecoratorExpressionToKind(Expression exp) {
+            switch (exp) {
+                case NameExpression ne when NameIsProperty(ne.Name):
+                case MemberExpression me when NameIsProperty(me.Name):
+                    return (SymbolKind.Property, FunctionKind.Property);
+
+                case NameExpression ne when NameIsStaticMethod(ne.Name):
+                case MemberExpression me when NameIsStaticMethod(me.Name):
+                    return (SymbolKind.Method, FunctionKind.StaticMethod);
+
+                case NameExpression ne when NameIsClassMethod(ne.Name):
+                case MemberExpression me when NameIsClassMethod(me.Name):
+                    return (SymbolKind.Method, FunctionKind.ClassMethod);
+            }
+
+            return null;
+        }
+
+        private bool NameIsProperty(string name) =>
+            name == "property"
+            || name == "abstractproperty"
+            || name == "classproperty"
+            || name == "abstractclassproperty";
+
+        private bool NameIsStaticMethod(string name) =>
+            name == "staticmethod"
+            || name == "abstractstaticmethod";
+
+        private bool NameIsClassMethod(string name) =>
+            name == "classmethod"
+            || name == "abstractclassmethod";
+
+        private void WalkAndDeclare(Node node) {
+            if (node == null) {
+                return;
+            }
+
+            _stack.EnterDeclared();
+            node.Walk(this);
+            _stack.ExitDeclaredAndMerge();
+        }
+
+        private void WalkAndDeclareAll(IEnumerable<Node> nodes) {
```

Comment:
Is it possible to call `WalkAndDeclareAll` with `nodes == null`?

Reply:
Yes. A try statement's array of handlers can be set to null by the parser.